### ISSUE TITLE
[Feature] Multithreading

### ISSUE THREAD

**Issue by simsong** [jessek/hashdeep#246]
Converted from [SourceForge issue 3120876](http://sourceforge.net/support/tracker.php?aid=3120876), submitted by nobody

It would be nice to have the capability of processing large folder structures in parallel using multi-threading. Maybe having a default value of number of processor cores detected with possibility to change the default behavior using a command line argument.
